### PR TITLE
Ujson and chunking

### LINKED ISSUE #181: Add ujson to data validator for data formatter df_to_json
The python json lib is much slower than other alternatives.

ujson is currently recognized as the faster encoder/decoder for python.  In testing, I saw the following results:
28785 entries - reduced JSON processing from about 0.52 seconds to 0.34 seconds
285285 entries - reduced JSON processing from about 7 seconds to 3 seconds
2850285 entries - reduced JSON processing from about 78 seconds to 58 seconds

### LINKED ISSUE #182: Chunk results dataframe for json processing
Really large dataframes, even when using ujson, can cause OOM errors because they're just too large.  Standard approach is to chunk up the dataframe for processing.

Right now I'm implementing this in the df_to_json formatter.  There may be other places where this will be needed, but so far it's here that I've run into errors when processing a really large (over 26 million) number of errors.   With chunking I was able to process that many errors in a little over 27 minutes.  I did a byte comparison of the json generated after chunking to json generated prior to chunking and the results were the same.

### PR DESCRIPTION
**Pull request by jcadam14**
Closes #181 
Closes #182

### CODE REVIEW

**Comment by jcadam14** [comment]
I put a comment in the dt_to_json to explain the reasoning behind doing a groupby before processing the error data.  Essentially this was the only way to maintain data integrity without adding even more processing to collate related data between chunks.  However, adding this groupby does impact smaller data sets, though not by a huge margin.  So if we think an absurd number of errors, in the 3 million+ is just not something we are going to concern ourselves with for now, I can remove all that and just keep ujson and the df.concat change.

Post MVP it might be looking at parallel processing across multiple services to really speed things up.

File: src/regtech_data_validator/data_formatters.py [hunk 1]
```diff
@@ -104,53 +102,54 @@ def df_to_table(df: pd.DataFrame) -> str:
...
-                    'severity': findings_json['validation_severity'][validation_key],
-                    'scope': findings_json['scope'][validation_key],
-                    'fig_link': findings_json['fig_link'][validation_key],
-                },
-                'records': [],
+    # grouping and processing keeps the process from crashing on really large error
+    # dataframes (millions of errors).  We can't chunk because could cause splitting
+    # related validation data across chunks, without having to add extra processing
+    # for tying those objects back together.  Grouping adds a little more processing
+    # time for smaller datasets but keeps really larger ones from crashing.
```
Comment (lchen-2101) [inline src/regtech_data_validator/data_formatters.py:109]: when you say crashing, does it kill the whole API with OOM or something like that? Or just the validation step fails? While "chunking" is worthwhile regardless, if just the validation fails, and doesn't kill the pod, then I'm less concerned with validation; maybe post-mvp we can add some sort of file has too many errors to process type of thing.

Comment (jcadam14) [reply src/regtech_data_validator/data_formatters.py:109]: https://github.com/cfpb/sbl-filing-api/pull/224 was written to handle if an executor crash happens during processing in the filing-api and gracefully move the submission to VALIDATION_ERROR

**Review by lchen-2101 — DISMISSED** [review]
LGTM

**Comment by lchen-2101** [comment]
oops, merge conflict needs resolving.

**Review by lchen-2101 — APPROVED** [review]
LGTM

**Pull request merged**